tests/test_finder.py

```diff
@@ -0,0 +1,90 @@
+import os
+import sys
+import shutil
+import zipfile
+import pytest
+from cx_Freeze import ModuleFinder, Module, ConstantsModule
+
+
+class TestModuleFinderWithConvertedNoseTests:
+    """ This class provides test cases that are conversions of the old NoseTests in `test_finder`
+    that predated usage of the PyTest Framework"""
+
+    @pytest.fixture()
+    def fix_module_finder(self):
+        constants = ConstantsModule()
+        mf = ModuleFinder(constants_module=constants)
+        return mf
+
+    def test_ScanCode(self, mocker, fix_test_samples_dir, fix_module_finder):
+        any3 = (mocker.ANY,) * 3
+        import_mock = mocker.patch.object(fix_module_finder, "_import_module", return_value=None)
+        fix_module_finder.IncludeFile(os.path.join(fix_test_samples_dir, "imports_sample.py"))
+        import_mock.assert_has_calls(
+            [
+                mocker.call("moda", *any3),
+                mocker.call("modb", *any3),
+                mocker.call("", *any3),
+                mocker.call("modd", *any3),
+                mocker.call("mode", *any3),
+                mocker.call("modf", *any3),
+                mocker.call("modg.submod", *any3),
+                mocker.call("modh", *any3),
+            ]
+        )
+
+    def test_not_import_invalid_module_name(self, mocker, fix_test_samples_dir, fix_module_finder):
+        """ testpkg1 contains not.importable.py, which shouldn't be included."""
+        fix_module_finder.path.insert(0, fix_test_samples_dir)
+        module = fix_module_finder.IncludePackage(
+            "testpkg1"
+        )  # Threw ImportError before the bug was fixed
+        assert ("invalid-identifier" in module.global_names), \
+            "submodules whose names contain invalid identifiers should still be imported"
+
+    def test_invalid_syntax(self, mocker, fix_test_samples_dir):
+        """Invalid syntax (e.g. Py2 or Py3 only code) should not break freezing."""
+        constants = ConstantsModule()
+        mf = ModuleFinder(path=[fix_test_samples_dir] + sys.path, constants_module=constants)
+        with pytest.raises(ImportError):
+            mf.IncludeModule(
+                "invalid_syntax"
+            )  # Threw SyntaxError before the bug was fixed
+
+    @pytest.mark.skip("Test skipped, uncertain if no longer supported - waiting for maintainer to comment on:"
+                      "https://github.com/marcelotduarte/cx_Freeze/pull/1234")
+    def test_FindModule_from_zip(self, fix_module_finder, fix_samples_dir, tmpdir):
```

Comment:
This test can be marked as skipped for now. In the rewrite of ModuleFinder, I ignored this method, but similar support for importing directing from the wheel (.whl)... Maybe we can delete it. I'll think about it

Reply:
Your shout mate :) - right now its there and disabled, if you want to nuke subsequently go for it :)

Reply:
Your shout - the current usage of `@pytest.mark.skip` disables execution of the test - if you want to remove the underlying code + test go for it. I didn't want to remove existing tests without discussion with you :)